src/sentry/services/hybrid_cloud/organization_mapping/impl.py

```diff
@@ -23,17 +24,19 @@ def create(
         # There's only a customer_id when updating an org slug
         customer_id: Optional[str] = None,
         user: Optional[int] = None,
+        status: Optional[OrganizationStatus] = None,
     ) -> RpcOrganizationMapping:
 
         if idempotency_key:
             org_mapping, _created = OrganizationMapping.objects.update_or_create(
                 slug=slug,
                 idempotency_key=idempotency_key,
                 region_name=region_name,
+                organization_id=organization_id,
                 defaults={
                     "customer_id": customer_id,
-                    "organization_id": organization_id,
                     "name": name,
+                    "status": status,
```

Comment:
If `status` is None, I think we want to default to `OrganizationStatus.ACTIVE`

Reply:
Hmm so I talked this over with @mikejihbe a bit yesterday, and we discussed the possibility of there being some intermediary state for organization. Since this can represent a slug reservation, wouldn't the more accurate status be `None` to signify that the org isn't actually active?

That being said, I suppose we're pulling status off the org in the controller code in every path so it'd likely always be active already 🤔

Reply:
`status` is a really, terrible name.  `status=ACTIVE` literally only means "not deleted".  Our async deletion logic depends on this. 

`status` is really used as "not being deleted".  I think it's always wise to create into status=ACTIVE.